pkg/routeagent/controllers/route/route_test.go

```diff
@@ -0,0 +1,46 @@
+package routecontroller
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("RouteController", func() {
+
+	Describe("Function populateCidrBlockList", func() {
+
+		Context("subnets does not contain CIDR blocks", func() {
+			It("should append CIDR block to subnets", func() {
+				routeController := RouteController{subnets: []string{"192.168.1.0/24"}}
+				routeController.populateCidrBlockList([]string{"10.10.10.0/24", "192.168.1.0/24"})
+				want := []string{"192.168.1.0/24", "10.10.10.0/24"}
+				Expect(routeController.subnets).To(Equal(want))
+			})
+		})
+
+		Context("subnets contain CIDR block", func() {
+			It("Should not append CIDR block subnets", func() {
```

Comment:
Similarly

    Context("When input CIDR blocks are already present in the existing subnets", func() {
        It("should not append the CIDR blocks to subnets", func() {